src/main/kotlin/com/dprint/listeners/OptimizeJavaImports.kt

```diff
@@ -0,0 +1,83 @@
+package com.dprint.listeners
+
+import com.dprint.config.ProjectConfiguration
+import com.dprint.core.Bundle
+import com.dprint.core.LogUtils
+import com.dprint.services.editorservice.EditorServiceManager
+import com.dprint.services.editorservice.FormatResult
+import com.intellij.lang.ImportOptimizer
+import com.intellij.lang.ImportOptimizer.CollectingInfoRunnable
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiJavaFile
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit
+
+private val LOGGER = logger<OptimizeJavaImports>()
+private const val TIMEOUT = 10L
+
+/**
+ * This listener formats Java files when optimize imports called.
+ */
+class OptimizeJavaImports : ImportOptimizer {
+
+    override fun supports(file: PsiFile): Boolean {
+        if (file !is PsiJavaFile) return false
+
+        val project = file.project
+        if (!project.service<ProjectConfiguration>().state.enabled) return false
+
+        val documentManager = PsiDocumentManager.getInstance(project)
+        val document = documentManager.getDocument(file) ?: return false
+
+        val editorServiceManager = project.service<EditorServiceManager>()
+        val fileManager = FileDocumentManager.getInstance()
+        return fileManager.getFile(document)?.path?.let { editorServiceManager.canFormatCached(it) } ?: false
+    }
+
+    override fun processFile(file: PsiFile): Runnable {
```

Comment:
Can we just add a `LogUtils.info` to this and maybe the `supports ` method that indicate we are using this listener. I have been trying to keep a pattern to that so that any help requests or bugs are easy to track down with minimal extra information from the reporters.